### PR TITLE
feat: implemented test timer

### PR DESCRIPTION
**Pull request by MilosNesh**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -382,3 +382,5 @@ _metadata.json
 
 /.angular/cache/
 *.db
+
+*/.angular/cache/
```
Comment (laslorajda) [inline .gitignore:386]: ?

File: Konteh/Konteh.FrontOfficeApi/Features/Exams/SubmitExam.cs [hunk 2]
```diff
@@ -37,6 +37,9 @@ public async Task Handle(Command request, CancellationToken cancellationToken)
             if (exam == null)
                 throw new NotFoundException();
 
+            if (exam.StartTime.AddMinutes(4) < DateTime.UtcNow) // set a test time limit (think about the time it takes to confirm submit + 30-60sec)
+                throw new NotFoundException();
```
Comment (laslorajda) [inline Konteh/Konteh.FrontOfficeApi/Features/Exams/SubmitExam.cs:41]: In all these scenarios I don't think a 404 is the correct response, but we can deal with that a different time.

File: Konteh/front-office-app/src/app/submit-dialog/submit-dialog.component.html [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<h2 mat-dialog-title>Submit</h2>
+<mat-dialog-content>
+  Time is up! Do you want to submit your test?
+</mat-dialog-content>
+<mat-dialog-actions>
+  <button mat-button mat-dialog-close>No</button>
```
Comment (laslorajda) [inline Konteh/front-office-app/src/app/submit-dialog/submit-dialog.component.html:6]: What does this button do 😄

File: Konteh/front-office-app/src/app/taking-test/taking-test.component.ts [hunk 4]
```diff
@@ -10,29 +12,37 @@ import { AnswerDto, ExamClient, GetExamQuestionDto, GetExamResponse, IGetExamRes
...
-        this.examQuestions = res;
-        this.question = res[0].questionDto!;
+        this.examQuestions = res.examQuestionDtos!;
+        this.question = res.examQuestionDtos![0].questionDto!;
+        this.endTime = new Date(res.startTime?.getTime()!)
+        this.endTime.setSeconds(this.endTime.getSeconds()+180) // set a test time limit
       },
-      error:()=>{
+      error:(err)=>{
+        console.log(err);
```
Comment (laslorajda) [inline Konteh/front-office-app/src/app/taking-test/taking-test.component.ts:34]: no console logs in repo!

**Review by zelick — APPROVED** [review]

**Pull request merged**